pkg/autodiscovery/providers/env_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package providers
```

Comment:
Tests for invalid JSON string and invalid logs config are missing

Reply:
That's the responsibility of the `Scheduler`, here we simply collect the config and send them through the AD